test/integration/filters/handle_socket_fail_filter.cc

```diff
@@ -0,0 +1,43 @@
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "source/common/network/connection_impl.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/extensions/filters/http/common/empty_http_filter_config.h"
+
+namespace Envoy {
+
+class OnSocketFailFilter : public Http::PassThroughFilter {
+public:
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool) override {
+    close(fd());
```

Comment:
Add comment on why this doesn't call ioHandle()->close() (so that IoHandle doesn't know the fd is invalid).